### PR TITLE
Side menu changes

### PR DESCRIPTION
**Pull request by rinat-shemer**
- add header with label and container for actions
- change the prefix to icon, use normal color as default
- change icon color on select

* Need to make changes in Bob where using this component (docs)

### CODE REVIEW

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu.stories.ts [hunk 1]
```diff
@@ -44,6 +47,7 @@ const note = `
   #### Properties
   Name | Type | Description
   --- | --- | --- | ---
+  headerLabel | string | header of menu 
   options | SideMenuOption[] | array of options
   onSelectOption | Action | select option emitter
```
Comment (galloween) [inline projects/ui-framework/src/lib/navigation/side-menu/side-menu.stories.ts]: - we have this new way of writing: [inputProp], [outputProp], so lets update
- need to add: [selectedId] | number | selected menu item index
- update the output to: ```(selectOption) | EventEmitter&lt;number&gt; | emits on option select ```

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.html [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
      (mouseenter)="setShowActions(true)"
      (mouseleave)="setShowActions(false)">
-  <div class="option-prefix">
-    <ng-template #prefix></ng-template>
-  </div>
+  <b-icon *ngIf="option.icon"
+          [icon]="option.icon"
+          [color]="selected ? colors.white : colors.normal"></b-icon>
   <div class="option-display-name"
        matTooltip="{{option.displayName}}"
```
Comment (galloween) [inline projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.html]: I dont like that it always displays tooltip, event if the item text fits fine.
lets replace this div with:

```
<b-truncate-tooltip [maxLines]="1" class="option-display-name">{{option.displayName}}</b-truncate-tooltip>
```

(you need to import TruncateTooltipModule and remove matTooltip import)

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.scss [hunk 3]
```diff
@@ -40,4 +31,8 @@
       text-align: left;
     }
   }
+
```
Comment (galloween) [inline projects/ui-framework/src/lib/navigation/side-menu/side-menu-option/side-menu-option.component.scss]: remove
```
      overflow: hidden;
      text-overflow: ellipsis;
      justify-self: auto;
      white-space: nowrap;
```

from .option-display-name

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu.module.ts [hunk 4]
```diff
@@ -18,6 +19,7 @@ import { ButtonsModule } from '../../buttons/buttons.module';
     MenuModule,
     MatTooltipModule,
```
Comment (galloween) [inline projects/ui-framework/src/lib/navigation/side-menu/side-menu.module.ts]: remove MatTooltipModule, add TruncateTooltipModule

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu.stories.ts [hunk 5]
```diff
@@ -16,14 +16,17 @@ import { IconComponent } from '../../icons/icon.component';
...
 import { SideMenuOption } from './side-menu-option/side-menu-option.interface';
+import { ButtonsModule } from 'bob-style';
 
 const inputStories = storiesOf(ComponentGroupType.Navigation, module)
   .addDecorator(withNotes)
   .addDecorator(withKnobs);
 
 const template = `
 <b-side-menu [options]="options"
+             [headerLabel]="headerLabel"
```
Comment (galloween) [inline projects/ui-framework/src/lib/navigation/side-menu/side-menu.stories.ts]: lets add [selectedId]="selectedId"

you will also need to add in the end of the file in the Props object:

selectedId: number('selectedId', 3)

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu.stories.ts [hunk 6]
```diff
@@ -52,14 +56,15 @@ const note = `
...
 
 inputStories.add(
   'Side Menu',
   () => {
     return {
       template: storyTemplate,
       props: {
+        headerLabel: text('headerLabel', 'test menu'),
         options: object<SideMenuOption[]>('options', sideMenuOptionsMock),
         selectOption: action('Side menu select')
```
Comment (galloween) [inline projects/ui-framework/src/lib/navigation/side-menu/side-menu.stories.ts]: add here: ```selectedId: number('selectedId', 3)```

File: projects/ui-framework/src/lib/navigation/side-menu/side-menu.component.html [hunk 7]
```diff
@@ -1,7 +1,12 @@
 <div class="side-menu-container">
```
[the remark this answers is not in the document]
Comment (rinat-shemer) [reply projects/ui-framework/src/lib/navigation/side-menu/side-menu.component.html]: it has `box-sizing: border-box;` without it the header lose it's width

**Pull request merged**